### DISCUSSION TITLE
Full screen reading

### DISCUSSION THREAD

**Discussion by chrisbd** [jocmp/capyreader#335] [Feature Requests]
Would it be possible that when you scroll down the article the top bar slides away to give a full screen reading experience?

**Comment by jocmp** [comment]
Hey @chrisbd. This should be straightforward, I'll try and add a setting in the next week. See https://github.com/jocmp/capyreader/issues/339

**Reply by sozifa** [reply]
As it has been implemented in the latest update - any chance you could add a setting for this as I'd prefer to have the top bar visible when scrolling

**Reply by jocmp** [reply]
@sozifa yep definitely. I'll keep the pinned top bar as the default. This will be an option under the settings.

<img src="https://github.com/user-attachments/assets/e2800510-6a10-4359-8d3d-15506dff39d9" width="300px">

**Comment by jocmp** [comment]
Closing since this feature was added in [2024.09.1049](https://github.com/jocmp/capyreader/releases/tag/2024.09.1049).

**Reply by chrisbd** [reply]
Thank you for implementing this feature so quickly! Amazing work. I wanted to share some feedback. When this feature is enabled, the top bar seems to appear too easily while scrolling through articles. It can be distracting because it often appears even when I'm not intentionally scrolling up in an article.

I've noticed that other reading apps like Omnivore have similar functionality, but I don't have this issue there. Perhaps the top bar's trigger is a bit too sensitive?

**Reply by jocmp** [reply]
thanks @chrisbd, I see what you mean. I tried out Omnivore (neat app!) too to compare.

This used a custom animation that only observed the last scroll which made it too touchy. I reworked the screen so that I can use the built-in Material behavior instead (https://github.com/jocmp/capyreader/pull/380). I'll make sure to add the change to the next release. Here's a sample of the before and after. Let me know what you think!

| Before  | After |
|--|--|
| <video src="https://github.com/user-attachments/assets/66d83fb5-ae0e-4588-aec3-c190eb7fb672"> | <video src="https://github.com/user-attachments/assets/63c9ffe6-2d1d-435c-a476-02a4938ac90a"> |

**Reply by chrisbd** [reply]
This looks much better! Nice one turning that around so quick. Will play around with it once I have the update and let you know. Thanks!
